engine/dlib/src/test/wscript

```diff
@@ -204,3 +206,14 @@ def build(bld):
     create_test(bld, 'test_opaque_handle_container')
     create_test(bld, 'test_crypt')
     create_test(bld, 'test_job_thread')
+
+    if bld.env.DOTNET:
+        b = bld.stlib(features= 'cs_stlib',
+                      project = 'cs/test_dlib_cs.csproj')
+        b.install_path = None
+
+        bld.add_group() # Until the cs_stlib can work with the 'use' mechanism properly
+        t = create_test(bld, 'test_csharp', extra_libs = ['CSHARP'])
+        t.env.append_value('LINKFLAGS', [t.env.LIB_ST % 'test_dlib_cs'])
+        t.env.append_value('LIBPATH', ['./src/test/cs'])
```

Comment:
Optional build (i.e. only if DOTNET is set)

Also, the waf rules aren't super easy to setup, so some manual setup is required.